contracts/kernel/organs/AccountingApp.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity ^0.4.8;                                                         
+
+contract AccountingApp {
+    
+    // The concept of sending tokens from one address to another
+    struct Transaction {
+        address token;
+        int value;
+        address baseToken;
+        address baseValue;
+        
+        address from;
+        address to;
+        string reference;
+
+        uint period;
+        // Back reference the states?
+        uint[] transactionStates;
+    }    
+    
+    // initial idea for states 
+    // TODO: draw out state machine
+    enum TransactionState {
+        New,
+        Approved,
+        Pending,
+        Canceled,
+        Failed,
+        Succeeded
+    }
+        
+    
+    struct TransactionAction {
+        uint transactionId;
```

Comment:
If we end up doing the mapping thing explained above, we can remove the id here and save 1 sstore per state change :)